### DISCUSSION TITLE
decorators in model

### DISCUSSION THREAD

**Discussion by sasha172017** [adonisjs/core#1873] [Help]
Is inpossible make custom decorator for property in model? And do pass  hookbeforeSave or beforeFind in custom decorator? this be would very nice, thank)

for example
@upload('folder')
@translate('ru')
@column()
public name: string

**Comment by thetutlage** [comment]
This has to do with Adonis. Decorators is an Javascript/Typescript thing and by following the typescript docs on decorators, you should be able to create and use them

**Reply by sasha172017** [reply]
But if i use getters/setters in my custom decorator than doesn't work setters/getters @column, does fix it?
